### PR TITLE
API Interface with a Few Routes

### PR DESCRIPTION
**Pull request by bh2smith**
We introduce an `iZerionApi` and `ZerionApi` which implements the interface. This includes implementations of three routes (`getChains`, `getPortfolio` and `getFungiblePositions`) and deeply nested response types.

### CODE REVIEW

File: src/types/fungible-position.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  // TODO
+  // changes: null | unknown;
+  fungible_info: FungibleInfo;
+  flags: PositionFlags;
+  updated_at: string; // ISO date string
+  updated_at_block: number;
+}
+
+export interface PositionQuantity {
+  int: bigint; // Large integer in string format
```
Comment (bh2smith) [inline src/types/fungible-position.ts:32]: Might be to transport strings here (for frontend client<>server data sharing).

Comment (tifrel) [reply src/types/fungible-position.ts:32]: In theory (ignoring JS precision issues), should this be true ? `
(int / 10 ** decimals) === float === parseFloat(numeric)
`

File: src/interface.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { ChainData, PortfolioData, PositionData } from "./types";
+
+// Interface for the Zerion API
+export interface iZerionAPI {
```
Comment (tifrel) [inline src/interface.ts:4]: I think the standard is to capitalize the `I` for interface types.

File: src/client.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  const response = await fetch(`${BASE_URL}${endpoint}`, { headers });
+
+  if (!response.ok) {
+    throw new Error(`Failed to fetch ${endpoint}: ${response.statusText}`);
+  }
+  const data = await response.json();
+  return data as T;
+}
+
+export class ZerionAPI implements iZerionAPI {
```
Comment (tifrel) [inline src/client.ts:35]: Why does this class need an interface?

**Review by tifrel — APPROVED** [review]

File: src/types/fungible-position.ts (hunk 1 above)
Comment (bh2smith) [reply src/types/fungible-position.ts:32]: I prefer to keep the bigint values until they need to be displayed in the interface. 

However I don't know what PositionQuantity is here (never really looked at the values contained in there).

File: src/client.ts (hunk 3 above)
Comment (bh2smith) [reply src/client.ts:35]: no good reason. I could move all the doc strings from the interface into the implementation and delete this. It was really just a blueprint for me to get started.

File: src/types/fungible-position.ts (hunk 1 above)
Comment (bh2smith) [reply src/types/fungible-position.ts]: Maybe float is best here. But we wont use this for a while so I will just make it a string again for now.

File: src/client.ts (hunk 3 above)
Comment (bh2smith) [reply src/client.ts]: No more interface.

**Pull request merged**